Fiber is failed if http4s request is cancelled

I faced with an issue that an error message appears in logs saying that fiber is failed when the client request is canceled. Another similar issue was fixed in pr #113.

I am expecting that the cancellation of requests should not produce an error message in logs and assuming that an issue in `zio.interop.CatsConcurrentEffect#runCancelable`, though I'm not sure.

I tested this with cats-effect IO and the error was not logged.
I also tested it with ZIO-RC17 and the behavior was the same as with RC18 (the error was logged).

Libraries versions:
```
org.http4s::http4s-blaze-server:0.21.1
org.http4s::http4s-dsl:0.21.1
dev.zio::zio:1.0.0-RC18
dev.zio::zio-interop-cats:2.0.0.0-RC11
```

Here's the sample code:
```scala
object MainTest extends CatsApp {
  override def run(args: List[String]) =
    BlazeServerBuilder[Task]
      .bindHttp(8030, "0.0.0.0")
      .withHttpApp(
        HttpRoutes
          .of[Task] { case _ => Task(Thread.sleep(100000)) *> Task(Response(Status.Ok)) }
          .orNotFound
      )
      .serve
      .compile
      .drain
      .as(0)
      .orElse(ZIO.succeed(1))
}
```
With cancelled get request:
```
❯ curl "http://localhost:8030"
^C⏎
```

Produces the following logs:
```
[DEBUG] 2020-03-06 14:10:26.146 SelectorLoop - Channel initialized.
[INFO ] 2020-03-06 14:10:26.149 BlazeServerBuilder - 
  _   _   _        _ _
 | |_| |_| |_ _ __| | | ___
 | ' \  _|  _| '_ \_  _(_-<
 |_||_\__|\__| .__/ |_|/__/
             |_|
[INFO ] 2020-03-06 14:10:26.207 BlazeServerBuilder - http4s v0.21.1 on blaze v0.14.11 started at http://[::]:8030/
[DEBUG] 2020-03-06 14:10:35.727 SelectorLoop - Channel initialized.
[DEBUG] 2020-03-06 14:10:35.728 NIO1HeadStage - Starting up.
[DEBUG] 2020-03-06 14:10:35.728 NIO1HeadStage - Stage NIO1HeadStage sending inbound command: Connected
[DEBUG] 2020-03-06 14:10:35.728 Http1ServerStage$$anon$1 - Starting HTTP pipeline
[DEBUG] 2020-03-06 14:10:35.730 IdleTimeoutStage - Starting idle timeout stage with timeout of 60000 ms
[DEBUG] 2020-03-06 14:11:35.803 IdleTimeoutStage - Idle timeout after 60000 ms.
[DEBUG] 2020-03-06 14:11:35.803 Http1ServerStage$$anon$1 - Shutting down due to idle timeout
[DEBUG] 2020-03-06 14:11:35.804 Http1ServerStage$$anon$1 - Shutting down HttpPipeline
[DEBUG] 2020-03-06 14:11:35.813 Http1ServerStage$$anon$1 - Shutting down.
[DEBUG] 2020-03-06 14:11:35.813 IdleTimeoutStage - Shutting down idle timeout stage
[DEBUG] 2020-03-06 14:11:35.813 IdleTimeoutStage - Shutting down.
[DEBUG] 2020-03-06 14:11:35.813 NIO1HeadStage - Shutting down.
[DEBUG] 2020-03-06 14:11:35.814 IdleTimeoutStage - Shutting down idle timeout stage
[DEBUG] 2020-03-06 14:11:35.814 IdleTimeoutStage - Shutting down.
[DEBUG] 2020-03-06 14:11:35.814 IdleTimeoutStage - Removed mid-stage: IdleTimeoutStage
[DEBUG] 2020-03-06 14:11:35.815 NIO1HeadStage - Stage NIO1HeadStage sending inbound command: Disconnected
[DEBUG] 2020-03-06 14:11:35.815 Http1ServerStage$$anon$1 - Shutting down HttpPipeline
[DEBUG] 2020-03-06 14:11:35.816 Http1ServerStage$$anon$1 - Shutting down.
[ERROR] 2020-03-06 14:11:35.861 Http1ServerStage$$anon$1 - Error running request: Request(method=GET, uri=/, headers=Headers(Host: localhost:8030, User-Agent: curl/7.64.1, Accept: */*))
zio.FiberFailure: Fiber failed.
╥
╠─An interrupt was produced by #32.
║ No ZIO Trace available.
║
╠─An interrupt was produced by #33.
║ No ZIO Trace available.
▼
[DEBUG] 2020-03-06 14:11:35.875 Http1ServerStage$$anon$1 - closeConnection()
[DEBUG] 2020-03-06 14:11:35.875 Http1ServerStage$$anon$1 - Shutting down HttpPipeline
[DEBUG] 2020-03-06 14:11:35.875 Http1ServerStage$$anon$1 - Shutting down.
[DEBUG] 2020-03-06 14:11:35.875 Http1ServerStage$$anon$1 - Shutting down HttpPipeline
[DEBUG] 2020-03-06 14:11:35.875 Http1ServerStage$$anon$1 - Shutting down.
[DEBUG] 2020-03-06 14:11:35.875 NIO1HeadStage - Shutting down.
[DEBUG] 2020-03-06 14:11:35.876 NIO1HeadStage - Stage NIO1HeadStage sending inbound command: Disconnected
[DEBUG] 2020-03-06 14:11:35.876 Http1ServerStage$$anon$1 - Shutting down HttpPipeline
[DEBUG] 2020-03-06 14:11:35.876 Http1ServerStage$$anon$1 - Shutting down.

```

--- a/interop-cats/shared/src/main/scala/zio/interop/cats.scala
+++ b/interop-cats/shared/src/main/scala/zio/interop/cats.scala
@@ -175,12 +175,15 @@ private class CatsConcurrentEffect[R](rts: Runtime[R])
   ): effect.SyncIO[effect.CancelToken[RIO[R, *]]] =
     effect.SyncIO {
       rts.unsafeRun {
-        RIO.unit
+        RIO.descriptor
           .bracketExit(
-            (_, exit: Exit[Throwable, A]) =>
+            (descriptor, exit: Exit[Throwable, A]) =>
               RIO.effectTotal {
-                if (exit.interrupted) ()
-                else effect.IO.suspend(cb(exit.toEither)).unsafeRunAsync(_ => ())
+                exit match {
+                  case Exit.Failure(cause) if !cause.interruptors.forall(_ == descriptor.id) => ()
+                  case _ =>
+                    effect.IO.suspend(cb(exit.toEither)).unsafeRunAsync(_ => ())
+                }
               },
             _ => fa
           )

--- a/interop-cats/shared/src/main/scala/zio/interop/cats.scala
+++ b/interop-cats/shared/src/main/scala/zio/interop/cats.scala
@@ -166,8 +166,7 @@ private class CatsConcurrentEffect[R](rts: Runtime[R])
   ): effect.SyncIO[Unit] =
     effect.SyncIO {
       rts.unsafeRunAsync(fa.run) { exit =>
-        if (exit.interrupted) ()
-        else cb(exit.flatMap(identity).toEither).unsafeRunAsync(_ => ())
+        cb(exit.flatMap(identity).toEither).unsafeRunAsync(_ => ())
       }
     }
 

--- a/interop-cats/shared/src/main/scala/zio/interop/cats.scala
+++ b/interop-cats/shared/src/main/scala/zio/interop/cats.scala
@@ -166,7 +166,8 @@ private class CatsConcurrentEffect[R](rts: Runtime[R])
   ): effect.SyncIO[Unit] =
     effect.SyncIO {
       rts.unsafeRunAsync(fa.run) { exit =>
-        cb(exit.flatMap(identity).toEither).unsafeRunAsync(_ => ())
+        if (exit.interrupted) ()
+        else cb(exit.flatMap(identity).toEither).unsafeRunAsync(_ => ())
       }
     }
 
@@ -179,7 +180,8 @@ private class CatsConcurrentEffect[R](rts: Runtime[R])
           .bracketExit(
             (_, exit: Exit[Throwable, A]) =>
               RIO.effectTotal {
-                effect.IO.suspend(cb(exit.toEither)).unsafeRunAsync(_ => ())
+                if (exit.interrupted) ()
+                else effect.IO.suspend(cb(exit.toEither)).unsafeRunAsync(_ => ())
               },
             _ => fa
           )

